src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +1026,265 @@ def _n_to_index_(self, n):
         except OverflowError:
             raise ValueError('value {} of index is negative'.format(n)) from None
 
+    def guess(self, f, n_max=100, max_dimension=10, sequence=None):
+        r"""
+        Guess a `k`-regular sequence of `(f(n))_{n\geq0}`.
+
+        INPUT:
+
+        - ``f`` -- a function (callable) which determines the sequence.
+          It takes nonnegative integers as an input
+
+        - ``n_max`` -- (default: ``100``) a positive integer. The resulting
+          `k`-regular sequence coincides with `f` on the first ``n_max``
+          terms
+
+        - ``max_dimension`` -- (default: ``10``) a positive integer specifying
+          the maxium dimension which is tried when guessing the sequence
+
+        - ``sequence`` -- (default: ``None``) a `k`-regular sequence used
+          for bootstrapping this guessing
+
+        OUTPUT:
+
+        A :class:`kRegularSequence`
+
+        EXAMPLES:
```

Comment:
it would be helpful to indicate the vector valued sequences used in the various examples (Sum of digits, variant 1: (s(n), s(2n+1)); variant 2: (1, s(n)), all natural numbers: (n, 2n+1), indicator function of even integers: ([n even], 1), indicator function of odd integers: )

https://github.com/sagemath/sage/issues/21204#issuecomment-1417820841 Item 8